From ad5339cfc277fc704147cea0e85541ec7254cbfb Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Thu, 27 Mar 2025 23:28:38 +0800
Subject: [PATCH] build: Remove superfluous header in compile command

PGO build fails because the compiler is trying to generate a header
based on the given header file. In fact, the header file shouldn't need
to be passed to the command, as it is already listed in the target's
dependency and all source files that include it should find it just
fine.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 main.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.mk b/main.mk
index 2803e623a2..b57190b7c0 100644
--- a/main.mk
+++ b/main.mk
@@ -1875,7 +1875,7 @@ xbin: sqltclsh$(T.exe) sqlite3_analyzer$(T.exe)
 
 sqlite3_expert$(T.exe): $(TOP)/ext/expert/sqlite3expert.h $(TOP)/ext/expert/sqlite3expert.c \
                        $(TOP)/ext/expert/expert.c sqlite3.c
-	$(T.link) $(TOP)/ext/expert/sqlite3expert.h $(TOP)/ext/expert/sqlite3expert.c \
+	$(T.link) $(TOP)/ext/expert/sqlite3expert.c \
 		$(TOP)/ext/expert/expert.c sqlite3.c -o sqlite3_expert $(LDFLAGS.libsqlite3)
 xbin: sqlite3_expert$(T.exe)
 
-- 
2.49.0

